components/match2/wikis/heroes/match_group_input_custom.lua

```diff
@@ -0,0 +1,660 @@
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
+		opponents[winnerIndex].status = _STATUS_DEFAULT_WIN
+		match.finished = true
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	match.finished = Logic.readBool(match.finished)
+		or isScoreSet and (
+			Array.any(opponents, function(opponent) return tonumber(opponent.score or 0) > match.bestof/2 end)
+			or Array.all(opponents, function(opponent) return tonumber(opponent.score or 0) == match.bestof/2 end)
+		)
+
+	-- see if match should actually be finished if score is set
+	if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+		local currentUnixTime = os.time(os.date('!*t') --[[@as osdate]])
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', match.date))
+		local threshold = match.dateexact and _SECONDS_UNTIL_FINISHED_EXACT
+			or _SECONDS_UNTIL_FINISHED_NOT_EXACT
+		if matchUnixTime + threshold < currentUnixTime then
+			match.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	if
+		not Logic.isEmpty(match.winner) or
+		Logic.readBool(match.finished) or
+		CustomMatchGroupInput.placementCheckSpecialStatus(opponents)
+	then
+		match.finished = true
+		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+function matchFunctions._makeAllOpponentsLoseByWalkover(opponents, walkoverType)
+	for index, _ in pairs(opponents) do
+		opponents[index].score = _NOT_PLAYED_SCORE
+		opponents[index].status = walkoverType
+	end
+	return opponents
+end
+
+-- Get Playerdata from Vars (get's set in TeamCards)
+function matchFunctions.getPlayersOfTeam(match, oppIndex, teamName, playersData)
+	-- match._storePlayers will break after the first empty player. let's make sure we don't leave any gaps.
+	playersData = Json.parseIfString(playersData) or {}
+	local players = {}
+	for playerIndex = 1, _MAX_NUM_PLAYERS do
+		-- parse player
+		local player = Json.parseIfString(match['opponent' .. oppIndex .. '_p' .. playerIndex]) or {}
+		player.name = player.name or playersData['p' .. playerIndex]
+			or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.flag = player.flag or playersData['p' .. playerIndex .. 'flag']
+			or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'flag')
+		player.displayname = player.displayname or playersData['p' .. playerIndex .. 'dn']
+			or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'dn')
+
+		if String.isNotEmpty(player.name) then
+			player.name = mw.ext.TeamLiquidIntegration.resolve_redirect(player.name):gsub(' ', '_')
+		end
+
+		if not Table.isEmpty(player) then
+			table.insert(players, player)
+		end
+	end
+	match['opponent' .. oppIndex].match2players = players
+	return match
+end
```

Comment:
used (almost) identical on most wikis, should we move this to commons?
@Rathoz

Reply:
Sure

Reply:
Though I think some wikis do it somewhat differently, like Valorant and LoL. We may want to look at the format we save them in

Reply:
yeah i will look into what is used the most and what we can combine into commons
if we need different wiki vars i might make it so that the format for them can be passed optionally
will see